lib/modelExtends.js

```diff
@@ -27,13 +27,17 @@ module.exports = function(Sequelize) {
 
 			var globalOptions = sequelize.options.define || {};
 
+			//If underscored(All) is set to true as global Sequelize option we let it override the model option
+			var underscored = this.options.underscored || this.sequelize.options.underscored;
```

Comment:
This should be `var underscored = this.options.underscored === undefined ? this.options.underscored : this.sequelize.options.underscored;`

The current line will defer to `sequelize.options.underscore` if `model.options.underscored == false` which isn't right. `== false` means definitively no underscoring.